File: algorithms/R/Rearrange Spaces Between Words/Rearrange Spaces Between Words.py
class Solution(object):
    def reorderSpaces(self, text):
        word_list = text.split()
        words, spaces = len(word_list), text.count(" ")
        
        if words > 1:
            q, r = spaces//(words-1), spaces%(words-1)
            return (" " * q).join(word_list) + " " * r
        else:
            return "".join(word_list) + " " * spaces

File: scripts/algorithms/R/Rearrange Spaces Between Words/Rearrange Spaces Between Words.js
// Runtime: 77 ms (Top 77.31%) | Memory: 42.2 MB (Top 47.52%)
var reorderSpaces = function(text) {
    let arr = text.split(" ");
    let totalSpace = arr.length-1;
    arr = arr.filter(w => w !== '');
    let spaceBetween = arr.length > 1 ?
                    Math.floor(totalSpace / (arr.length-1)) : 0;
    let spaceLeftOver = arr.length > 1 ?
                    totalSpace % (arr.length-1) : totalSpace;
    return (arr.join(" ".repeat(spaceBetween)) + " ".repeat(spaceLeftOver));
    // Time Complexity: O(n)
    // Space Complexity: O(n)
};